doc/.ChangeLog_template

```diff
@@ -48,45 +52,42 @@ Changes made to namelist defaults (e.g., changed parameter values):
 
 Changes to the datasets (e.g., parameter, surface or initial files):
 
-Substantial timing or memory changes: [For timing changes, can check PFS test(s) in the test suite]
+Substantial timing or memory changes:
+[For timing changes, you should at least check the PFS test in the test
+ suite, whether or not you expect timing changes: ensure that its run
+ time has not increased significantly relative to the baseline.]
+
 
-Notes of particular relevance for developers: (including Code reviews and testing)
+Notes of particular relevance for developers:
 ---------------------------------------------
 NOTE: Be sure to review the steps in README.CHECKLIST.master_tags as well as the coding style in the Developers Guide
+[Remove any lines that don't apply. Remove entire section if nothing applies.]
 
 Caveats for developers (e.g., code that is duplicated that requires double maintenance):
 
 Changes to tests or testing:
 
-CTSM testing:
```

Comment:
I'd actually like to keep this section. I find it a useful reminder on what to test, and I always remove it before making a tag.

Reply:
This section has never made sense to me and has felt redundant with the list of possible tests below. Note that I did try to add some of the relevant information from here to the list of tests below. Can you explain what value this section has for you beyond what is already available in the list of tests below?

Reply:
Obviously we need it to make sense to everyone making tags. This really came out of a CLM software discussion from years ago when we worked out that it was sometimes acceptable to NOT always run all of the testing. At that time we worked out some sample use cases when you wouldn't run all of the testing. The following is basically the table of those use cases. What I find useful with the table is to identify the testing level for a particular PR and then do that level of testing and report it that way. So it helps me to identify what tests I should be doing as well as what I should report on. And it also lets me feel like it's legit to not run all the testing for each tag. Otherwise, I tend to feel like I should always run all the testing. I'd expect it to be useful to others making tags in the same way.

Now, technically you are right that it's redundant in that you then report on what testing you actually run in the parts below. So if we agree that it's acceptable to not always run all the testing every tag, maybe it's not really needed. But, again I find it a useful table for me to figure out what level of testing I should do for a tag.

Reply:
I remember having that discussion, but I've felt that the use cases listed here don't really apply anymore: Have we ever had a tag in the last few years for which we've just run the 'short' testing, for example; and is there ever a case where you would run the build-namelist testing but not the system tests? And on the other hand, this makes no mention of when to run ptclm or python testing. It also hasn't made sense to me to have distinct "levels"; for example, what would the "test level" be if I changed the tools and some obscure part of the python but nothing else?

I do still agree that there are (rare) cases when it makes sense not to run all of the testing – for example, if you just update something in the python directory that has no impact on most / any system runs; or if you just change documentation.

My feeling is that the best way to do this is to try to give rules for when each of the (currently five) categories of testing should be run. I have tried to do that for build-namelist,tools-tests and python testing. I'm not sure what the rule should be for PTCLM. We could also add a rule for regular tests, though I think it would be more subjective, like "should be run in nearly all cases, unless changes have no impact on system runs".

All that said: In the end, it's not a big deal to me if we keep this in if it's providing some real value for you. I can continue to routinely ignore it and just delete it when I make each tag.

Reply:
Let's hear from @negin513 and @slevisconsulting to see if the table is useful to them.

I have done dead simple tags where I just run the short test or maybe one specific test. Very rare though as you point out. I find it hard to imagine a case where you would run the build-namelist testing -- but NOT the system tests. But, that could be the case if you were just fixing the build-namelist testing and didn't do anything that would affect any of the system tests. I think I've had that happen before.

PTCLM is being deprecated, but how about I made a stab at editing your PR to add a note around it? I've always thought of that test as something that I'm the only one that actually does it, but it's useful for me to see when I do. It's easy enough to edit the file directly in the PR if you are OK with my doing that.

Reply:
Yup, please go ahead and make any changes you'd like.

Reply:
I think this table has been useful for me to find the level of testing (especially at the beginning); however, I do agree with @billsacks that there are not any tags in the past year for which we've just run the `short` testing or `build-namelist` testing.

Overall, I think this includes useful information and we should keep it. Everyone should remove this paragraph before making a tag, so it does not cause any clutters.  I agree with @ekluzek to keep this in the template as a guide.

Reply:
@negin513 - to be honest, your statement that this was useful to you in the beginning is to me an argument that this should be removed or at least reworked: I feel like this is more misleading than helpful to a newcomer, since it gives the impression that there are different allowable levels of testing, whereas in fact nearly all of our tags need to be tested at the "standard" level.

That said, if it's useful to you and @ekluzek I'm still okay keeping it. But can one of you please rewrite it so that it reflects the current reality. Feel free to edit the text I've added in the list below describing when each category should be run in order to keep these two sets of instructions consistent.

Reply:
> My feeling is that the best way to do this is to try to give rules for when each of the (currently five) categories of testing should be run.

I support this idea.

Reply:
OK, so I reworked it -- feel free to comment.

This is a section that I've used and I've labeled the test level for my tags. As such I could go back and see that about 10% of those tags were something other than "regular". There was also about that amount where tools testing was done along with regular testing. I did have a few tags with short, but I changed the level description to "minimal" so you could then describe what you did. There were also a few with None or N/A that would probably now be done labeled as "doc" level.

I also removed the "build-namelist" level as that could really just be described by a special case of the "minimal" level (which in itself is really a special case category). Personally I run the build-namelist tests for every tag, because they are short and it's easy for me to fix them. So I never labeled any tags that I just ran that level.

Reply:
I also added a ":" to the header for the section, as I plan to continue to add the level to my tags after the header.  It's mostly useful if you are doing something besides "regular", so I don't think we should require it. But, maybe recommend it when you are doing something besides "regular".

Reply:
Can everyone comment if my changes are acceptable? If so I'll resolve this conversation.